bokehjs/src/lib/models/tools/inspectors/hover_tool.ts

```diff
@@ -249,183 +251,149 @@ export class HoverToolView extends InspectToolView {
     const tooltips: [number, number, HTMLElement | null][] = []
 
     if (glyph instanceof PatchView) {
-      const [rx, ry] = [sx, sy]
+      const [tt_sx, tt_sy] = [sx, sy]
+      const [tt_x, tt_y] = [x, y]
       const vars = {
-        x, y, sx, sy, rx, ry,
+        index: null,
+        glyph, x, y, sx, sy, tt_x, tt_y, tt_sx, tt_sy,
         name: renderer.name,
       }
-      tooltips.push([rx, ry, this._render_tooltips(ds, -1, vars)])
+      const rendered = this._render_tooltips(ds, vars)
+      tooltips.push([tt_sx, tt_sy, rendered])
     }
 
-    if (glyph instanceof HAreaView) {
+    if (glyph instanceof VAreaView || glyph instanceof HAreaView) {
       for (const i of subset_indices.line_indices) {
-        const data_x1 = glyph._x1
-        const data_x2 = glyph._x2
-        const data_y = glyph._y
-        const [rx, ry] = [sx, sy]
+        const [tt_x, tt_y] = [x, y]
+        const [tt_sx, tt_sy] = [sx, sy]
         const vars = {
           index: i,
-          x, y, sx, sy, data_x1, data_x2, data_y, rx, ry,
-          indices: subset_indices.line_indices,
+          glyph, x, y, sx, sy, tt_x, tt_y, tt_sx, tt_sy,
           name: renderer.name,
-        }
-        tooltips.push([rx, ry, this._render_tooltips(ds, i, vars)])
-      }
-    }
-
-    if (glyph instanceof VAreaView) {
-      for (const i of subset_indices.line_indices) {
-        const data_x = glyph._x
-        const data_y1 = glyph._y1
-        const data_y2 = glyph._y2
-        const [rx, ry] = [sx, sy]
-        const vars = {
-          index: i,
-          x, y, sx, sy, data_x, data_y1, data_y2, rx, ry,
           indices: subset_indices.line_indices,
-          name: renderer.name,
         }
-        tooltips.push([rx, ry, this._render_tooltips(ds, i, vars)])
+        const rendered = this._render_tooltips(ds, vars)
+        tooltips.push([tt_sx, tt_sy, rendered])
       }
     }
 
     if (glyph instanceof LineView) {
+      const {line_policy} = this.model
       for (const i of subset_indices.line_indices) {
-        let data_x = glyph._x[i+1]
-        let data_y = glyph._y[i+1]
-        let ii = i
-
-        let rx: number
-        let ry: number
-        switch (this.model.line_policy) {
-          case "interp": { // and renderer.get_interpolation_hit?
-            [data_x, data_y] = glyph.get_interpolation_hit(i, geometry)
-            rx = xscale.compute(data_x)
-            ry = yscale.compute(data_y)
-            break
+        const [[tt_x, tt_y], [tt_sx, tt_sy], ii] = (function() {
```

Comment:
We use arrow functions as an idiom in this context. Using function is technically permissible, but may indicate that `this` will be treated in a special way, which I don't see to be the case here. Another sad exception are generator functions which do not have arrow equivalent.

Reply:
[60556ad](https://github.com/bokeh/bokeh/pull/12147/commits/60556ad9f4b1a9a25bf12298df52dccc99ded769)